restcomm/restcomm.telephony.api/src/main/java/org/restcomm/connect/telephony/api/util/CallControlHelper.java

```diff
@@ -63,10 +64,15 @@ static boolean permitted(final String authorization, final String method, DaoMan
         final String response = map.get("response");
         final ClientsDao clients = daoManager.getClientsDao();
         final Client client = clients.getClient(user, organizationSid);
-        //only allow if client algo is identical to system algo
+
+        if (client != null && daoManager.getAccountsDao().getAccount(client.getAccountSid()).getStatus() != Account.Status.ACTIVE) {
+            return false;
+        }
+
         if (client != null && Client.ENABLED == client.getStatus()) {
-            final String password2 = client.getPassword();
-            final String result = DigestAuthentication.response(algorithm, user, realm, "", password2, nonce, nc, cnonce,
+            final String password = client.getPassword();
+            final String clientPasswordAlgorithm = client.getPasswordAlgorithm();
+            final String result = DigestAuthentication.response(authHeaderAlgorithm, user, realm, password,clientPasswordAlgorithm, nonce, nc, cnonce,
```

Comment:
will this work when "cleartext" is populated?

Reply:
@jaimecasero yes, its already covered in the testsuite